org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentSymbolHandler.java

```diff
@@ -43,6 +44,9 @@
 	private void collectChildren(ITypeRoot unit, IJavaElement[] elements, ArrayList<SymbolInformation> symbols)
 			throws JavaModelException {
 		for(IJavaElement element : elements ){
+			if (element.getElementName() == null || element.getElementName().startsWith("lambda$") || element.getElementName().startsWith("$")) {

```

Comment:
The only two methods I know are:
org.eclipse.jdt.core.IMethod.isLambdaMethod()
org.eclipse.jdt.core.IType.isLambda()